frontend/pluginloader.lua

```diff
@@ -1,42 +1,66 @@
 local lfs = require("libs/libkoreader-lfs")
 local logger = require("logger")
 
-local PluginLoader = {
-    plugin_path = "plugins"
-}
+local DEFAULT_PLUGIN_PATH = "plugins"
+
+local PluginLoader = {}
 
 function PluginLoader:loadPlugins()
     if self.plugins then return self.plugins end
 
     self.plugins = {}
-    for f in lfs.dir(self.plugin_path) do
-        local path = self.plugin_path.."/"..f
-        local mode = lfs.attributes(path, "mode")
-        -- valid koreader plugin directory
-        if mode == "directory" and f:find(".+%.koplugin$") then
-            local mainfile = path.."/".."main.lua"
-            local package_path = package.path
-            local package_cpath = package.cpath
-            package.path = path.."/?.lua;"..package.path
-            package.cpath = path.."/lib/?.so;"..package.cpath
-            local ok, plugin_module = pcall(dofile, mainfile)
-            if not ok or not plugin_module then
-                logger.warn("Error when loading", mainfile, plugin_module)
-            elseif type(plugin_module.disabled) ~= "boolean" or not plugin_module.disabled then
+    local lookup_path_list = { DEFAULT_PLUGIN_PATH }
+    local extra_paths = G_reader_settings:readSetting("extra_plugin_paths")
```

Comment:
I actually don't really expect many people to use this feature right now. We can always add a default path to the list if it turns out to be a popular feature. My preference is to not introduce this extra lookup overhead to all users for now.